docs/narrative.rst

```diff
@@ -75,10 +75,24 @@ Example when using the layer:
     ...     def make_wsgi_app(self):
     ...         return zope.testbrowser.wsgi.AuthorizationMiddleware(simple_app)
 
-There is also a BrowserLayer in `zope.app.wsgi.testlayer`_ which does this
+There is also a ``BrowserLayer`` in `zope.app.wsgi.testlayer`_ which does this
 for you and includes a ``TransactionMiddleware``, too, which could be handy
 when testing a ZODB based application.
 
+However, since the ``BrowserLayer`` in `zope.app.wsgi.testlayer`_ re-creates
+the ZODB in ``testSetUp``, we need to re-create the WSGI App during
+``testSetUp``, too. Therefore use ``TestBrowserLayer`` of
+``zope.testbrowser.wsgi`` instead of the simpler ``Layer`` to combine it with
+the ``BrowserLayer`` in `zope.app.wsgi.testlayer`_:
+
+.. doctest::
+
+    >>> import zope.testbrowser.wsgi
+    >>> import zope.app.wsgi.testlayer
+    >>> class Layer(zope.testbrowser.wsgi.TestBrowserLayer,
+    ...             zope.app.wsgi.testlayer.BrowserLayer):
+    ...     pass
+
```

Comment:
Other Zope layers do not support composition via `plone.testing.Layer` (as far as I understand it). And since `zope.testbrowser.wsgi.TestBrowserLayer` relies on the `make_wsgi_app` implementation of `zope.app.wsgi.testlayer.BrowserLayer` I see no other way than multiple inheritance.

I'm not that happy myself, but since it works, it's a start. Maybe after we have done more migrations we can come up with a better solution.